src/App.tsx

```diff
@@ -1,61 +1,133 @@
+import { Todo } from './components/types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App = () => {
+  const [copyTodos, setCopyTodos] = useState([...todosFromServer]);
+  const [selectedUserName, setSelectedUserName] = useState('');
+  const [newTitle, setNewTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const maxId = Math.max(...todosFromServer.map(todo => +todo.id));
+  const findUser = (userName: string): User | null => {
+    return usersFromServer.find(user => user.name === userName) || null;
+  };
+
+  const checkError = () => {
+    if (!newTitle) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserName) {
+      setErrorUser(true);
+    }
+  };
+
+  const createTodo = () => {
+    const user = findUser(selectedUserName);
+
+    if (user && newTitle) {
+      const newTodo = {
+        id: maxId,
+        title: newTitle,
+        completed: false,
+        userId: user.id,
+      };
+
+      setCopyTodos(todos => [...todos, newTodo]);
+      setNewTitle('');
+      setSelectedUserName('');
+    }
+
+    return null;
+  };
+
+  const addHandler
+  = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    createTodo();
+    checkError();
+  };
+
+  const todos: Todo[] = copyTodos.map(todo => {
+    return {
+      ...todo,
+      user: usersFromServer.find(user => todo.userId === user.id),
+    };
+  });
+
+  useEffect(() => {
+    if (newTitle) {
+      setErrorTitle(false);
+    }
+
+    if (selectedUserName) {
+      setErrorUser(false);
+    }
+  });
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTitle}
+              onChange={(e) => {
+                setNewTitle(e.currentTarget.value);
+              }}
+            />
+          </label>
+          {(errorTitle
+             && <span className="error">Please enter a title</span>)}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
+          <label>
+            User:
+            <select
+              data-cy="userSelect"
+              onChange={(e) => setSelectedUserName(e.target.value)}
+            >
+              {!selectedUserName
+               && (<option value="0" disabled selected>Choose a user</option>)}
```

Comment:
The 'selected' attribute in the 'option' element is not recommended in React. Use the 'value' attribute in 'select' tag instead.